### PR TITLE
fix crafting ignoring most checks

### PR DESCRIPTION
**Pull request by Xander3359**
## About The Pull Request
Gonna have to port this back to TG:
- Fix crafting ignoring most checks
- Adds engineering skill check to crafting times (for things that require it)
## Why It's Good For The Game
bugfix
## Changelog
:cl:
fix: Fix crafting ignoring construction checks
/:cl:

### CODE REVIEW

File: code/datums/components/crafting/_recipes.dm [hunk 1]
```diff
@@ -31,6 +31,16 @@
...
+	/// Setting this to true will effectively set check_direction to true.
+	var/is_fulltile = FALSE
+	/// If this atom should run the direction check, for use when building things like directional windows where you can have more than one per turf
+	var/check_direction = FALSE
+	/// If the atom requires a floor below
+	var/on_solid_ground = FALSE
+	/// If the atom checks that there are objects with density in the same turf when being built. TRUE by default
+	var/check_density = TRUE
+	/// Bitflag of additional placement checks required to place. (STACK_CHECK_CARDINALS|STACK_CHECK_ADJACENT|STACK_CHECK_TRAM_FORBIDDEN|STACK_CHECK_TRAM_EXCLUSIVE)
+	var/placement_checks = NONE
```
Comment (Lumipharon) [inline code/datums/components/crafting/_recipes.dm:43]: at this point all these flags should be... flags tbh.

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/datums/components/crafting/_recipes.dm (hunk 1 above)
Comment (Xander3359) [reply code/datums/components/crafting/_recipes.dm:43]: 😅

File: code/datums/components/crafting/_recipes.dm [hunk 2]
```diff
@@ -41,6 +51,8 @@
 	var/result_amount
 	/// Whether we should delete the contents of the crafted storage item (Only works with storage items, used for ammo boxes, donut boxes, internals boxes, etc)
 	var/delete_contents = TRUE
+	///whether only people with sufficient construction skill can build this.
+	var/skill_req = FALSE
```
Comment (TiviPlus) [inline code/datums/components/crafting/_recipes.dm:55]: is this a bool? doesnt look like it to me?

File: code/datums/components/crafting/_recipes.dm (hunk 1 above)
Comment (TiviPlus) [reply code/datums/components/crafting/_recipes.dm:43]: we live in a bool society

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**